File: bridges/post-play-audio-advanced-bridge/post-play-audio-advanced.js
//Download the node sdk from ap.bandwidth.com/docs/helper-libraries/node-js
//API credentials which can be found on your account page at https://catapult.inetwork.com/pages/login.jsf
var userId = 'userId';  //{user_id}
var token = 'token'; //{token}
var secret = 'secret'; //{secret}

var Bandwidth = require('node-bandwidth');

var client = new Bandwidth({
	userId: userId,
	apiToken: token,
	apiSecret: secret
});

//Play Audio File on loop
var options = {
	fileUrl     : "http://myurl.com/file.mp3",
	loopEnabled : true
}
//Promise
client.Bridge.playAudioAdvanced("bridgeId", options)
.then(function (response) {
	console.log(response);
});

//Callback
client.Bridge.playAudioAdvanced("bridgeId", options, function (err, response) {
	if(err) {
		console.log(err);
	}
	else {
		console.log(response);
	}
});

//Speak sentence with options
var options = {
	sentence : "hola de Bandwidth",
	gender   : "male",
	locale   : "es",
	voice    : "Jorge"
}
//Promise
client.Bridge.playAudioAdvanced("bridgeId", options)
.then(function (response) {
	console.log(response);
});

//Callback
client.Bridge.playAudioAdvanced("bridgeId", options, function (err, response) {
	if(err) {
		console.log(err);
	}
	else {
		console.log(response);
	}
});

File: bridges/post-play-audio-advanced-bridge/PostPlayAudioAdvanced.cs
//Download the .net sdk from ap.bandwidth.com/docs/helper-libraries/net
using System;
using System.Threading.Tasks;
using Bandwidth.Net;

public class Program
{
  //API credentials which can be found on your account page at https://catapult.inetwork.com/pages/login.jsf
  private const string UserId = "u-userID";  //{user_id}
  private const string Token = "t-token"; //{token}
  private const string Secret = "secret"; //{secret}

  public static void Main()
  {
    try
    {
      RunAsync().Wait();
    }
    catch (Exception ex)
    {
      Console.Error.WriteLine(ex.Message);
      Environment.ExitCode = 1;
    }
  }

  private static async Task RunAsync()
  {
    var client = new Client(UserId, Token, Secret);

    //Play Audio File on loop
    await client.Bridge.PlayAudioAsync("bridgeId", new PlayAudioData
    {
      FileUrl = "http://myurl.com/file.mp3",
      LoopEnabled = true
    });

    //Speak sentence with options
    await client.Bridge.PlayAudioAsync("bridgeId", new PlayAudioData
    {
      Sentence = "hola de Bandwidth",
      Gender = Gender.Male,
      Locale = "es",
      Voice  = "Jorge"
    });
  }
}
